src/app/tools/symmetry.cpp

```diff
@@ -0,0 +1,95 @@
+// Aseprite
+// Copyright (C) 2021  Igara Studio S.A.
+//
+// This program is distributed under the terms of
+// the End-User License Agreement for Aseprite.
+
+#ifdef HAVE_CONFIG_H
+#include "config.h"
+#endif
+
+#include "app/tools/symmetry.h"
+
+#include "app/tools/point_shape.h"
+#include "app/tools/stroke.h"
+#include "app/tools/tool_loop.h"
+#include "doc/brush.h"
+
+namespace app {
+namespace tools {
+
+void Symmetry::generateStrokes(const Stroke& stroke, Strokes& strokes,
+                               ToolLoop* loop)
+{
+  Stroke stroke2;
+  strokes.push_back(stroke);
+  gen::SymmetryMode symmetryMode = loop->getSymmetry()->mode();
+  switch (symmetryMode)
+  {
+    case gen::SymmetryMode::NONE:
+      ASSERT(false);
+      break;
+
+    case gen::SymmetryMode::HORIZONTAL:
+    case gen::SymmetryMode::VERTICAL:
+
+      calculateSymmetricalStroke(stroke, stroke2, loop, symmetryMode);
+      strokes.push_back(stroke2);
+      break;
+
+    case gen::SymmetryMode::BOTH:
+      gen::SymmetryMode auxSymmetryMode = gen::SymmetryMode::HORIZONTAL;
+      calculateSymmetricalStroke(stroke, stroke2, loop, auxSymmetryMode);
+      strokes.push_back(stroke2);
+      auxSymmetryMode = gen::SymmetryMode::VERTICAL;
+      Stroke stroke3;
+      calculateSymmetricalStroke(stroke, stroke3, loop, auxSymmetryMode);
+      strokes.push_back(stroke3);
+      auxSymmetryMode = gen::SymmetryMode::BOTH;
+      Stroke stroke4;
+      calculateSymmetricalStroke(stroke3, stroke4, loop, auxSymmetryMode);
+      strokes.push_back(stroke4);
+      break;
```

Comment:
Several minor changeS:
```suggestion
    case gen::SymmetryMode::BOTH: {
      calculateSymmetricalStroke(stroke, stroke2, loop, gen::SymmetryMode::HORIZONTAL);
      strokes.push_back(stroke2);

      Stroke stroke3;
      calculateSymmetricalStroke(stroke, stroke3, loop, gen::SymmetryMode::VERTICAL);
      strokes.push_back(stroke3);

      Stroke stroke4;
      calculateSymmetricalStroke(stroke3, stroke4, loop, gen::SymmetryMode::BOTH);
      strokes.push_back(stroke4);
      break;
    }
```